lib/puppet/functions/gitlab_ci_runner/register_to_file.rb

```diff
@@ -41,6 +41,11 @@ def register_to_file(url, regtoken, runner_name, additional_options = {}, proxy
       return 'DUMMY-NOOP-TOKEN' if Puppet.settings[:noop]
 
       begin
+        # Confirm the specified ca file exists
+        if !ca_file.nil? && !File.exist?(ca_file)
+          puts('Specified CA file doesn\'t exist for gitlab-ci-runner. Did you forget to create it?')
```

Comment:
Should probably use `Puppet.warning` here instead of `puts`?

The message itself might be better hinting at possibly needing to run Puppet again???

```
Unable to register gitlab runner at this time as the specified `ca_file` does not exist (yet).  If puppet is managing this file, the next run should complete the registration process.
```

???

Reply:
Agreed, I have switched to using Puppet.warning and updated the message provided to the user.

Reply:
Did you forget to push your update?